scripts/application_to_provider.py

```diff
@@ -577,7 +580,11 @@ def migrate_image_data_glance(src_glance_client, dst_glance_client, img_uuid, lo
         logging.debug("Attempting to upload image data to destination provider")
         with open(local_path, 'rb') as img_file:
             try:
-                dst_glance_client.images.upload(img_uuid, img_file)
+                # "Upload" method is different for Glance client v1, than for v2
+                if isinstance(dst_glance_client, glanceclient.v1.client.Client):
```

Comment:
Sort of minor, did a bit of reading to confirm this, you have the option to do `type(dst_glance_client) is glanceclient.v1.client.Client`. If the v2 client was a subclass of the v1 client, then your check wouldn't work as intended. `isinstance` is typically preferred, because of duck typing (when you want to check if an instance has at minimum the behavior of some class and its parents), but in cases like this I think the preferred approach is to check for an exact class match.

Reply:
sound reasonable, change made as suggested